[19.0][MIG] base_exception: Migration to 19.0

lef-adhoc opened the pull request.

mourad-ehm committed:
[ADD] split module sale_exception into base_exception

lmignon committed:
[FIX] base_exception: '_check_exception' is 'api.multi' since it's called by constraint methods
'detect_exception' can be called on an empty recordset.

mourad-ehm committed:
[MIG] base_exception: Migration to 11.0

mourad-ehm committed:
[FIX] temp model test access rule & fix OCA remarks

astirpe committed:
[FIX] Enable post_install for tests + pylint

mourad-ehm committed:
[WIP] sale_exception_by_domain

ivantodorovich committed:
[IMP] Update base_exception views

- Show menu only to Exception Rule Managers
- Use sequence and active widgets on tree view
- Updated form to use sheet

bealdav committed:
FIX remove old api parameters in base_exception

nicomacr committed:
[FIX] base_exception: send the active_model in the conxtext when call the _popup_exceptions mehtod

ivantodorovich committed:
[MIG] base_exception: Migration to 12.0

ivantodorovich committed:
FIX self[0] in _rule_group

ivantodorovich committed:
[MIG] dict.iteritems() does not exist in Python3

Commit:
[IMP] Add convenience action to ignore exceptions on record

[IMP] Computed exception descriptions field, to display better help messages

[IMP] Exceptions shouldn't be copied

jaredkipe committed:
[IMP] base_exception: black, isort

jaredkipe committed:
[MIG] base_exception: Migration to 13.0

kevinkhao committed:
[IMP] base_exception: black, isort, prettify

kevinkhao committed:
[14.0][MIG] base_exception: Migration to 14.0

sebastienbeau committed:
[IMP] add method exception

sebastienbeau committed:
[REF] Refactor test

use odoo-test-helper
split test so one test do one thing

sebastienbeau committed:
[FIX] fix access right

JordiBForgeFlow committed:
[13.0][fix][base_exception] search only active exception rules

TDu committed:
[13.0][IMP] base_exception add blocking exceptions

By default exceptions can be ignored by the click of a button.
Users begin human they will just click that button what ever
the internal rules.

So this PR adds the option to set specific exceptions as blocking.

When exceptions are detected if one of them is blocking, the user
will not be able to ignore them.

MiquelRForgeFlow committed:
[FIX] base_exception: import correctly osv.expression

kittiu committed:
[14.0][FIX] base_exception, error if rule's description is null

ivantodorovich committed:
[IMP] base_exception: Allow to skip exception checks through context

hailangvn committed:
[IMP] base_exception: test blocking exceptions

ivantodorovich committed:
[IMP] base_exception: log traceback when code exception is raised

cesar-tecnativa committed:
[MIG] base_exception: Migration to 15.0

MiquelRForgeFlow committed:
[FIX] base_exception: add _popup_exceptions deleted in migration

This method is used by other modules that depend on 'base_exception'.

OCA-git-bot committed:
base_exception 15.0.1.0.1

CarlosRoca13 committed:
[FIX] base_exception: Fix tests and add odoo_test_helper as dependency

OCA-git-bot committed:
base_exception 15.0.1.0.2

matiasperalta1 committed:
[MIG] base_exception: Migration to 16.0

OCA-git-bot committed:
base_exception 16.0.1.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (10 of 60 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (10 of 60 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/it/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (53 of 53 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/es_AR/

aliciagaarzo committed:
[16.0][IMP] base_exception: Improved wizard view

mymage committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (10 of 60 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 18.8% (10 of 53 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/sl/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (10 of 60 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/it/

glitchov committed:
Translated using Weblate (French)

Currently translated at 86.7% (46 of 53 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/fr/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/

OCA-git-bot committed:
base_exception 16.0.1.0.2

lmignon committed:
[IMP] base_exception: One model per python file

lmignon committed:
[IMP] base_exception: Improves performances

Before this change, the inverse relation from the exception to the records linked to the exception was read to determine if the rule should be added or removed from the rule detected for the current records. On large database with non blocking exceptions this lead to a performance issues since this issues a read of all the SO where the exception has been applied. With this change, we only read the information from the current records

lmignon committed:
[IMP] base_exception: Improves performances

Keeps rule information into cache to avoid to always call the ORM for static informations. This is change allows a boost into the evaluation of exception rules by avoiding useless SQL queries to retrieve rules definitions

lmignon committed:
[IMP] base_exception: Improves performances

Don't call the ORM to filter current recordset. Replace call to the orm by a call to the filtered_domain method.

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (54 of 54 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/es/

OCA-git-bot committed:
base_exception 16.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/

lef-adhoc committed:
[IMP] base_exception: pre-commit auto fixes

lef-adhoc committed:
[MIG] base_exception: Migration to 17.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-base_exception/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-base_exception/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (51 of 51 strings)

Translation: server-tools-17.0/server-tools-17.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-base_exception/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (51 of 51 strings)

Translation: server-tools-17.0/server-tools-17.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-base_exception/es/

rousseldenis committed:
[IMP] base_exception: Add a hook method / Don't build void exceptions

In order to get exceptions without writing recordset, add a hook method.

Improve exceptions dictionaries building using a defaultdict of a browse
record.

xaviedoanhduy committed:
[MIG] base_exception: Migraion to 18.0

Commit:
[UPD] Update base_exception.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update base_exception.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-18.0/server-tools-18.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-base_exception/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (52 of 52 strings)

Translation: server-tools-18.0/server-tools-18.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-base_exception/it/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 15.3% (8 of 52 strings)

Translation: server-tools-18.0/server-tools-18.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-base_exception/nl/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 94.2% (49 of 52 strings)

Translation: server-tools-18.0/server-tools-18.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-base_exception/nl/

lef-adhoc commented:
The tests in the sale_exception module passed https://github.com/OCA/sale-workflow/pull/3944. We need to adapt odoo_test_helper. I think it's something like this: https://github.com/OCA/odoo-test-helper/pull/34

rousseldenis commented:
/ocabot migration base_exception

rousseldenis commented:
@lef-adhoc Could you try to include `odoo-test-helper` PR here in `test-requirements.txt` ?

rven commented:
> @lef-adhoc Could you try to include `odoo-test-helper` PR here in `test-requirements.txt` ?

I don't think the migration of `odoo-test-helper` for v19 is complete.
I stopped the migration, because it now can be handled by standard Odoo like mentioned in that PR

matiasperalta1 reviewed (approved):
LGTM

lef-adhoc committed:
[MIG] base_exception: Migration to 19.0

lef-adhoc commented:
Hi @rousseldenis 

PR https://github.com/OCA/odoo-test-helper/pull/34 was giving errors, so I've adapted odoo-test-helper for v19 and created PR https://github.com/OCA/odoo-test-helper/pull/35 with the necessary changes for the new model registration API.

Updated [test-requirements.txt](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) to use this PR.

etobella commented:
@lef-adhoc 

Can you try to remove odoo_test_helper and use the mechanism commented here?

https://github.com/OCA/odoo-test-helper/pull/34

lef-adhoc commented:
@etobella Done!
I based it on the migration of edi_core_oca and deleted odoo_test_helper.

rousseldenis commented on base_exception/tests/test_base_exception.py:
```diff
@@ -47,11 +54,6 @@ def setUpClass(cls):
 
-    @classmethod
-    def tearDownClass(cls):
-        cls.loader.restore_registry()
```
@lef-adhoc You should keep this but using:

```
 cls.addClassCleanup(cls.registry.__delitem__, "base.exception.test.purchase.line")

etc...
```

lef-adhoc commented:
@rousseldenis I removed the test_purchase_ids = fields.Many2many("base.exception.test.purchase") field from [ExceptionRule](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html). This field was creating a Many2many relationship from the [exception.rule](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) model to the test model [base.exception.test.purchase](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html). The problem occurred during teardown: when [addClassCleanup](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) deleted [base.exception.test.purchase](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) from the registry, the test_purchase_ids field became orphaned (referencing a non-existent model). This caused [AssertionError: Field exception.rule.test_purchase_ids with unknown comodel_name 'base.exception.test.purchase'](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) during [reset_changes()](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html). Since this field wasn't actually used in any test assertions, removing it was the cleanest solution.

Also added @mute_logger("odoo.registry") to suppress expected ERROR logs during test model registration. When [add_to_registry](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) registers models, Odoo's registry verification detects they don't have database tables yet and logs [ERROR odoo.registry: Model X has no table](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html). These errors are harmless (tables are created immediately after by [init_models()](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)), but the ERROR keyword in logs causes CI to fail. The decorator mutes these registry errors without affecting actual test failures.

rousseldenis commented:
@lef-adhoc @etobella So, the mechanism is not yet error prone safe.

etobella commented:
@rousseldenis no, it is safe. 

He was missing this 
```python
cls.addClassCleanup(cls.registry["exception.rule"]._fields.__delitem__, "test_purchase_ids")
```

I have tested it in local and worked like a charm :wink:

lef-adhoc commented:
@etobella I changed it as you said, but it's not working for me.
Is there something wrong?

etobella commented:
Myself... I didn't execute the full tests. Sorry.

However, there is a way for making it work

https://github.com/OCA/server-tools/commit/42e3025ecbc1b379d03e887669850808b82b4608

Can you test? it worked for me.

lef-adhoc commented:
@rousseldenis @etobella I think it's fine now

etobella reviewed (approved):
Thanks!

lef-adhoc commented:
@rousseldenis Could we merge?

sebalix commented:
/ocabot merge nobump
